modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -1002,6 +1002,59 @@ TEST_P(Imgcodecs_Tiff_Modes, decode_multipage_use_memory_buffer)
     }
 }
 
+TEST_P(Imgcodecs_Tiff_Modes, decode_multipage_use_memory_buffer_selected_pages)
+{
+    const int mode = GetParam();
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root + "readwrite/multipage.tif";
+    const string page_files[] = {
+        "readwrite/multipage_p1.tif",
+        "readwrite/multipage_p2.tif",
+        "readwrite/multipage_p3.tif",
+        "readwrite/multipage_p4.tif",
+        "readwrite/multipage_p5.tif",
+        "readwrite/multipage_p6.tif"
+    };
+    const size_t page_count = sizeof(page_files) / sizeof(page_files[0]);
+
+    FILE* fp = fopen(filename.c_str(), "rb");
+    ASSERT_TRUE(fp != NULL);
+    fseek(fp, 0, SEEK_END);
+    long pos = ftell(fp);
+
+    std::vector<uchar> buf;
+    buf.resize((size_t)pos);
+    fseek(fp, 0, SEEK_SET);
+    buf.resize(fread(&buf[0], 1, buf.size(), fp));
+    fclose(fp);
+
+    vector<Mat> pages_from_imread;
+    for (size_t i = 0; i < page_count; i++)
+    {
+        const Mat page = imread(root + page_files[i], mode);
+        pages_from_imread.push_back(page);
+    }
+
+    for (size_t start = 0; start < page_count; start++)
+    {
+        for (size_t end = start + 1; end <= page_count; end++)
+        {
+            vector<Mat> pages_from_imdecodemulti;
+            Range const range(start, end);
+            bool res = imdecodemulti(buf, mode, pages_from_imdecodemulti, range);
+            ASSERT_TRUE(res == true);
+            ASSERT_EQ(static_cast<size_t>(range.size()), pages_from_imdecodemulti.size());
+
+            for (int i = range.start; i < range.end; i++)
+            {
+                EXPECT_PRED_FORMAT2(cvtest::MatComparator(0, 0),
+                    pages_from_imread[i],
+                    pages_from_imdecodemulti[i - range.start]);
+            }
+        }
+    }
+}
```

Comment:
I do not think that we need all combination here. It's enough to call `imdecodemulti` once with trange like `(1, size-2)` to check not default behaviour. Other cases and tiff specific are indirectly tested with `imreadmulti`.

Reply:
I'll revise the code as requested, but please note that, strictly speaking, the proposed test does not check all combinations, which would be nonsensical, but all _contiguous subsequences_. If you have a sequence of length _n_, there are _n_ × (_n_ + 1)/2 ~ O(_n_<sup>2</sup>) contiguous subsequences, which for the current _n_ = 6 gives 21.